### PR TITLE
Linked list kth

### PR DESCRIPTION
**Pull request by catdude2000**
- [x] Top-level README “Table of Contents” is updated
 - [x] README for this challenge is complete
       - [ ] Summary, Description, Approach & Efficiency, Solution
       - [ ] Picture of whiteboard
       - [ ] Link to code
 - [x] Feature tasks for this challenge are completed
 - [x] Unit tests written and passing
       - [ ] “Happy Path” - Expected outcome
       - [ ] Expected failure
       - [ ] Edge Case (if applicable/obvious)

### CODE REVIEW

File: javascript/linked-list/linked-list-kth/kth.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  constructor(value, next = null) {
+    this.value = value;
+    this.next = next;
+  }
+}
+
+function kthFromEnd(head, k) {
+  if (!head || k <= 0) {
+    return null;
+  }
```
Comment (DSFrey) [inline javascript/linked-list/linked-list-kth/kth.js:11]: `k = 0` is a valid input (it's the last node in the list), so it should be
``` js
if(!head || k < 0)

File: javascript/linked-list/linked-list-kth/kth.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    if (!fastPointer) {
+      return null;
+    }
+    fastPointer = fastPointer.next;
+  }
+
+  while (fastPointer) {
+    slowPointer = slowPointer.next;
+    fastPointer = fastPointer.next;
+  }
```
Comment (DSFrey) [inline javascript/linked-list/linked-list-kth/kth.js:26]: We want to stop our loop when our fastPointer is on the last item, not after it's gone off the end. That means we should be checking `fastPointer.next`.

File: javascript/linked-list/linked-list-kth/kth.test.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
+const { kthFromEnd, createLinkedList } = require('./kth');
+
+describe('kthFromEnd', () => {
+  test('should return the kth node from the end', () => {
+    const head = createLinkedList([1, 2, 3, 4, 5, 6, 7]);
+    const kthNode = kthFromEnd(head, 3);
+    expect(kthNode.value).toBe(5);
+  });
```
Comment (DSFrey) [inline javascript/linked-list/linked-list-kth/kth.test.js:8]: As per the instructions, the last node is 0 from the end, so the node 3 from the end should be 4 not 5.

File: javascript/linked-list/linked-list-kth/kth.test.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    const head = createLinkedList([1, 2, 3, 4, 5]);
+    const kthNode = kthFromEnd(head, 7);
+    expect(kthNode).toBeNull();
+  });
+
+  test('should return null for an empty list', () => {
+    const head = createLinkedList([1, 2]);
+    const kthNode = kthFromEnd(head, 2);
+    expect(kthNode.value).toBe(1);
+  });
```
Comment (DSFrey) [inline javascript/linked-list/linked-list-kth/kth.test.js:20]: You are not testing what your test name says you are.
